test-framework/junit5/src/main/java/io/quarkus/test/junit/launcher/DockerContainerLauncherProvider.java

```diff
@@ -42,25 +47,73 @@ public DockerContainerArtifactLauncher create(CreateContext context) {
                 launcher = new DefaultDockerContainerLauncher();
             }
             SmallRyeConfig config = (SmallRyeConfig) LauncherUtil.installAndGetSomeConfig();
-            launcher.init(new DefaultDockerInitContext(
-                    config.getValue("quarkus.http.test-port", OptionalInt.class).orElse(DEFAULT_PORT),
-                    config.getValue("quarkus.http.test-ssl-port", OptionalInt.class).orElse(DEFAULT_HTTPS_PORT),
-                    TestConfigUtil.waitTimeValue(config),
-                    TestConfigUtil.integrationTestProfile(config),
-                    TestConfigUtil.argLineValue(config),
-                    TestConfigUtil.env(config),
-                    context.devServicesLaunchResult(),
-                    containerImage,
-                    pullRequired,
-                    additionalExposedPorts(config),
-                    labels(config),
-                    volumeMounts(config)));
+            launcherInit(context, launcher, config, containerImage, pullRequired, Optional.empty(), volumeMounts(config),
+                    Collections.emptyList());
             return launcher;
         } else {
```

Comment:
I would like have a general comment here describing on a high level what this block is  doing